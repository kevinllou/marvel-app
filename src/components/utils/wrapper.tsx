import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import MarvelRoutes from '../../routes/MarvelRoutes';

const renderWithBrowserRouter = (component: ReactElement) => render(
  <Provider store={store}>
    <BrowserRouter>{component}</BrowserRouter>
  </Provider>,
);

const renderWithMemoryRouter = (path: string[] = ['/']) => render(
  <Provider store={store}>
    <MemoryRouter initialEntries={path}>
      <MarvelRoutes />
    </MemoryRouter>
  </Provider>,
);

export {
  renderWithBrowserRouter,
  renderWithMemoryRouter,
};
