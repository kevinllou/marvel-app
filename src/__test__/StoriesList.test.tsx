import { waitFor } from '@testing-library/react';
import { renderWithBrowserRouter } from '../components/utils/wrapper';
import StoriesGrid from '../components/Story/StoryGrid';

describe('Stories List Component', () => {
  beforeEach(() => {
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });

  it('should render list of stories', async () => {
    const { container, getByText } = renderWithBrowserRouter(<StoriesGrid />);
    await waitFor(() => {
      expect(container.querySelector('.spinner')).not.toBeTruthy();
      expect(getByText(/Investigating the murder of a teenage girl/i)).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
});
