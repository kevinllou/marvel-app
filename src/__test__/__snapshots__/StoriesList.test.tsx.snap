// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories List Component should render list of stories 1`] = `
<div>
  <div
    class="characters__grid"
  >
    <div
      class="stories__gridItems"
    >
      <div
        class="stories__card"
      >
        <div
          class="stories__cardImage"
        >
          <img
            alt=""
            src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
          />
        </div>
        <div
          class="stories__cardBody"
        >
          <div
            class="stories__name"
          >
            <a
              href="/stories/7"
            >
              Investigating the murder of a teenage girl, Cage suddenly learns that a three-way gang war is under way for control of the turf
            </a>
          </div>
          <div
            class="stories__options"
          >
            <button
              style="color: white;"
              type="button"
            >
              <i
                class="fa-sharp fa-solid fa-heart"
              />
            </button>
            <button
              style="color: white; pointer-events: auto;"
              type="button"
            >
              <i
                class="fa-solid fa-eye-slash"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="stories__gridItems"
    >
      <div
        class="stories__card"
      >
        <div
          class="stories__cardImage"
        >
          <img
            alt=""
            src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
          />
        </div>
        <div
          class="stories__cardBody"
        >
          <div
            class="stories__name"
          >
            <a
              href="/stories/8"
            >
              In the wake of September 11th, the world watched as firefighters, police officers and EMT workers selflessly risked their lives
            </a>
          </div>
          <div
            class="stories__options"
          >
            <button
              style="color: white;"
              type="button"
            >
              <i
                class="fa-sharp fa-solid fa-heart"
              />
            </button>
            <button
              style="color: white; pointer-events: auto;"
              type="button"
            >
              <i
                class="fa-solid fa-eye-slash"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <section
    class="pagination"
  >
    <button
      class="pagination__prev"
      disabled=""
      type="button"
    >
      <i
        class="fa-solid fa-angle-left"
      />
    </button>
    <button
      class="pagination__btn pagination__btn--active"
      style="pointer-events: auto;"
      type="button"
    >
      1
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: auto;"
      type="button"
    >
      2
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: auto;"
      type="button"
    >
      3
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: auto;"
      type="button"
    >
      4
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: auto;"
      type="button"
    >
      5
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: none;"
      type="button"
    >
      ...
    </button>
    <button
      class="pagination__btn"
      style="pointer-events: auto;"
      type="button"
    >
      78
    </button>
    <button
      class="pagination__next"
      type="button"
    >
      <i
        class="fa-solid fa-angle-right"
      />
    </button>
  </section>
</div>
`;
